homeassistant/util/file.py

```diff
@@ -54,11 +54,11 @@ def write_utf8_file(
     """
 
     tmp_filename = ""
-    tmp_path = os.path.split(filename)[0]
+    file_dir = os.path.dirname(filename)
```

Comment:
I like dirname, but I'm not sure the renaming of the variable adds anything.

Reply:
I figured someone might ask about the rename. :D 

`path` can generally refer to a full pathname including a filename (see e.g. `pathlib.Path`), so it's a bit ambiguous. 

Additionally, the variable is not just "temporary path", it's literally the directory name of the file (even if in this context it is used for determining where to put a temporary file).

Reply:
Can you use a full name then?
`file_directory` or just `directory`

Reply:
Better yet, I just got rid of the name altogether, since the local was only used on (effectively) the following line.